datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs

```diff
@@ -92,6 +98,242 @@ pub(crate) fn prune_row_groups(
     filtered
 }
 
+/// Prune row groups by bloom filters
+///
+/// Returns a vector of indexes into `groups` which should be scanned.
+///
+/// If an index is NOT present in the returned Vec it means the
+/// predicate filtered all the row group.
+///
+/// If an index IS present in the returned Vec it means the predicate
+/// did not filter out that row group.
+pub(crate) async fn prune_row_groups_by_bloom_filters<
+    T: AsyncFileReader + Send + 'static,
+>(
+    builder: &mut ParquetRecordBatchStreamBuilder<T>,
+    row_groups: &[usize],
+    groups: &[RowGroupMetaData],
+    predicate: &PruningPredicate,
+    metrics: &ParquetFileMetrics,
+) -> Vec<usize> {
+    let bf_predicates = match BloomFilterPruningPredicate::try_new(predicate.orig_expr())
```

Comment:
IMO @hengfeiyang call `orig_expr ` will not get the `Inlist` rewrite here, maybe you should write a test for `select * from table col1 in (xxx)` 🤔

Reply:
BTW, this is a awesome feature, thx for your great work! 👍

Reply:
Sure, i will add a test for `col in(xxx)`

Reply:
BTW I have some ideas (working on them in https://github.com/alamb/arrow-datafusion/pull/14) of how to support `InList` expressions directly by using the interval analysis library.